ddtrace/llmobs/_integrations/crewai.py

```diff
@@ -0,0 +1,260 @@
+        if span.error:
+            return
+        span._set_ctx_item(OUTPUT_VALUE, response)
+
+    def _llmobs_set_tags_tool(self, span, args, kwargs, response):
+        tool_instance = kwargs.get("instance")
+        tool_name = getattr(tool_instance, "name", "")
+        description = _extract_tool_description_field(getattr(tool_instance, "description", ""))
+        span._set_ctx_items(
+            {
+                NAME: tool_name if tool_name else "CrewAI Tool",
+                METADATA: {"description": description},
+                INPUT_VALUE: kwargs.get("input", ""),
+            }
+        )
+        if span.error:
+            return
+        span._set_ctx_item(OUTPUT_VALUE, response)
+
+    def _llmobs_set_span_link_on_task(self, span, queued_task, finished_task_outputs, crew_instance):
+        """Set span links for the next queued task in a CrewAI workflow.
+        This happens between task executions, (the current span is the crew span and the task span hasn't started yet)
+        so we create span links to be set on the task span once it starts later.
+        We rely on 3 cases to determine the appropriate span links:
+        1. queued_task.context is set with the most recently finished tasks that directly feed into the queued task.
+        2. queued_task.context is empty and there are no finished task outputs,
+            meaning this is the first task (tasks if async) in the crew workflow.
+        3. queued_task.context is empty, but there are n finished task outputs,
+            meaning that the last n task outputs should be the pre-requisite tasks for the queued task.
+        """
+        if not self.llmobs_enabled or not self.span_linking_enabled:
+            return
+        queued_task_id = getattr(queued_task, "id", "")
+        is_planning_crew_instance = span.trace_id in self._planning_crew_traces
+        queued_task_node = self._traces_to_tasks[span.trace_id].setdefault(str(queued_task_id), {})
+        span_links = []
+
+        if queued_task.context:
+            for finished_task in queued_task.context:
+                finished_task_id = getattr(finished_task, "id", "")
+                finished_task_node = self._traces_to_tasks[span.trace_id].get(str(finished_task_id), {})
+                finished_task_span_id = finished_task_node.get("span_id")
+                span_links.append(
+                    {
+                        "span_id": finished_task_span_id,
+                        "trace_id": "{:x}".format(span.trace_id),
+                        "attributes": {"from": "output", "to": "input"},
+                    }
+                )
+            queued_task_node["span_links"] = span_links
+            return
+        if not finished_task_outputs and not is_planning_crew_instance:
+            queued_task_node["span_links"] = [
+                {
+                    "span_id": str(span.span_id) if span else ROOT_PARENT_ID,
+                    "trace_id": "{:x}".format(span.trace_id),
```

Comment:
should we also make this conditional on if span exists, as we do above? and similarly in other spots in the function